### PR TITLE
⚠️ Log according to sig-instrumentation guidelines

### PR DESCRIPTION
**Pull request by vincepri**
**What type of PR is this?**

**What this PR does / why we need it**:

/kind cleanup

See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#what-method-to-use

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
If you're using --verbosity (or `-v`) flags to gather more logs, the new guidelines show: warn at level 1, debug at level 4, and trace at level 5.
```

### CODE REVIEW

**Comment by Skarlso** [comment]
I have a fix for your failing unit test https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4579

File: pkg/logger/logger.go [hunk 1]
```diff
@@ -43,43 +45,52 @@ type Wrapper interface {
 
 // Logger is a concrete logger using logr underneath.
 type Logger struct {
+	helper func()
```
Comment (AndiDog) [inline pkg/logger/logger.go:48]:
```suggestion
	callStackHelper func()
```

**Review by AndiDog — COMMENTED** [review]
/lgtm

One minor naming remark.

**Review by AndiDog — COMMENTED** [review]
/lgtm

**Pull request merged**